addons/point_of_sale/static/src/js/Screens/ClientListScreen/ClientDetailsEdit.js

```diff
@@ -49,8 +49,18 @@ odoo.define('point_of_sale.ClientDetailsEdit', function(require) {
                     processedChanges[key] = value;
                 }
             }
-            if ((!this.props.partner.name && !processedChanges.name) ||
-                processedChanges.name === '' ){
+
+            let savedName = this.props.partner.name;
+            let newName = processedChanges.name;
+            let isMissingName = false;
+            // if invalid name in db  (undefined or empty/blank string)
+            if (!savedName || (savedName && savedName.trim() === '') ) 
+                isMissingName = true;
+            // if new input, check that that it is valid
+            if (typeof newName === 'string') 
```

Comment:
I think this condition should be `!isMissingName && typeof newName === 'string'`.

Reply:
We always need to check the `typeof newName === 'string'` condition. Because even if `isMissingName` is true from the previous "if" (which means bad savedName in db) maybe the user inserted a valid `newName`. So if we skip this "if" it will be like "not recognizing" the newly inserted data. What do you think?

Reply:
You're right. :)